### PR TITLE
feat: upgrade to Bevy 0.10.

### PR DESCRIPTION
**Pull request by lecoqjacob**

### CODE REVIEW

**Review by zicklag — COMMENTED** [review]
Awesome, and a big thank you!! It doesn't look like it ended up too bad.

File: crates/bones_bevy_renderer/CHANGELOG.md
[the remark this answers is not in the document]
Comment (lecoqjacob) [reply crates/bones_bevy_renderer/CHANGELOG.md:1]: yeah ill remove that. i was going to make a pass at the changelogs, but it seems you have an automated system for that

File: crates/quinn_runtime_bevy/Cargo.toml [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
-bevy_tasks = "0.9.1"
-async-executor = "1.4.1"
-async-io = "1.9.0"
-pin-project = "1.0.12"
-futures-lite = "1.12.0"
+async-executor = "1.4"
+async-io       = "1.9"
+bevy_tasks     = "0.10"
+futures-lite   = "1"
+pin-project    = "1"
```
Comment (zicklag) [inline crates/quinn_runtime_bevy/Cargo.toml:15]: Is there a reason to make these version specifications less specific?

I'm not necessarily against it, but I was wondering if it was intentional.

As far as I understand, doing `1` is the same as `1.0.0`, and that would assert that we are compatible with `1.0.0` exactly, and everything after `1.0`.

That's probably true, but in order to be sure, we would have to try and build the game with version `1.0.0` exactly and make sure it still works. I've actually only ever tested the game with whatever was the latest-released version of the crate at the time I added it, so I only knew that the more specific specifier was actually tested.

**Comment by zicklag** [comment]
I just pushed a commit to fix the documentation issue in CI. I think this is ready as soon as I get your feedback on the version numbers. 🚀

**Pull request merged**